hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -1138,6 +1138,61 @@ vlapic_calcdest(struct acrn_vm *vm, uint64_t *dmask, uint32_t dest, bool phys, b
 	}
 }
 
+/*
+ * This function populates 'dmask' with the set of vcpus that is the possible destination
+ * when lapic is passthru.
+ */
+void
+vlapic_calcdest_lapic_pt(struct acrn_vm *vm, uint64_t *dmask, uint32_t dest, bool phys)
+{
+	struct acrn_vlapic *vlapic;
+	uint64_t amask;
+	uint32_t ldr, logical_id, dest_logical_id, dest_cluster_id;
+	uint16_t vcpu_id;
+
+	if (dest == 0xffU) {
+		/*
+		 * Broadcast in both logical and physical modes.
+		 */
+		*dmask = vm_active_cpus(vm);
+	} else if (phys) {
+		/*
+		 * Physical mode: destination is LAPIC ID.
+		 */
+		*dmask = 0UL;
+		vcpu_id = vm_apicid2vcpu_id(vm, (uint8_t)dest);
```

Comment:
2nd parameter of vm_apicid2vcpu_id should be changed to uint32_t with X2APIC. This is not your issue, can you fix too?

Reply:
Sure, I will fix and update.